pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -128,345 +154,465 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
-	if !externalSecret.Status.RefreshTime.IsZero() {
-		timeSinceLastRefresh = time.Since(externalSecret.Status.RefreshTime.Time)
-	}
-
 	// skip reconciliation if deletion timestamp is set on external secret
-	if externalSecret.DeletionTimestamp != nil {
-		log.Info("skipping as it is in deletion")
+	if !externalSecret.GetDeletionTimestamp().IsZero() {
+		log.V(1).Info("external secret is marked for deletion, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
 	// if extended metrics is enabled, refine the time series vector
 	resourceLabels = ctrlmetrics.RefineLabels(resourceLabels, externalSecret.Labels)
 
 	if shouldSkipClusterSecretStore(r, externalSecret) {
-		log.Info("skipping cluster secret store as it is disabled")
+		log.V(1).Info("skipping cluster secret store as it is disabled")
 		return ctrl.Result{}, nil
 	}
 
 	// skip when pointing to an unmanaged store
 	skip, err := shouldSkipUnmanagedStore(ctx, req.Namespace, r, externalSecret)
 	if skip {
-		log.Info("skipping unmanaged store as it points to a unmanaged controllerClass")
+		log.V(1).Info("skipping unmanaged store as it points to a unmanaged controllerClass")
 		return ctrl.Result{}, nil
 	}
 
-	refreshInt := r.RequeueInterval
-	if externalSecret.Spec.RefreshInterval != nil {
-		refreshInt = externalSecret.Spec.RefreshInterval.Duration
-	}
-
 	// Target Secret Name should default to the ExternalSecret name if not explicitly specified
 	secretName := externalSecret.Spec.Target.Name
 	if secretName == "" {
-		secretName = externalSecret.ObjectMeta.Name
+		secretName = externalSecret.Name
+	}
+
+	// fetch the existing secret (from the partial cache)
+	//  - please note that the ~partial cache~ is different from the ~full cache~
+	//    so there can be race conditions between the two caches
+	//  - the WatchesMetadata(v1.Secret{}) in SetupWithManager() is using the partial cache
+	//    so we might receive a reconcile request before the full cache is updated
+	//  - furthermore, when `--enable-managed-secrets-caching` is true, the full cache
+	//    will ONLY include secrets with the "managed" label, so we cant use the full cache
+	//    to reliably determine if a secret exists or not
+	secretPartial := &metav1.PartialObjectMetadata{}
+	secretPartial.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("Secret"))
+	err = r.Get(ctx, client.ObjectKey{Name: secretName, Namespace: externalSecret.Namespace}, secretPartial)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, logErrorGetSecret, "secretName", secretName, "secretNamespace", externalSecret.Namespace)
+		syncCallsError.With(resourceLabels).Inc()
+		return ctrl.Result{}, err
 	}
 
-	// fetch external secret, we need to ensure that it exists, and it's hashmap corresponds
-	var existingSecret v1.Secret
-	err = r.Get(ctx, types.NamespacedName{
-		Name:      secretName,
-		Namespace: externalSecret.Namespace,
-	}, &existingSecret)
+	// if the secret exists but does not have the "managed" label, add the label
```

Comment:
Why do we take ownership of a secret that we don't manage? 🤔

Reply:
So if the target exists, but we don't own it, we will override it and own it. And then if the caching is enabled, we will use the special secret client to fetch the just owned secret from the cache. 

That will be a cache miss at that point, right? And so, we do our thing, it gets cached and from there, we are fine. Did I get that right?

Reply:
You are correct, if we find that an ExternalSecret is targeting an existing Secret without the "managed" label, we apply the label so that it's visible in the cached secrets client, and immediately re-queue the reconcile.

It's not so much "taking ownership", but marking that we care about this secret by adding a label.